### ISSUE TITLE
A Touching story

### ISSUE THREAD

**Issue by timothyallan** [tannerlinsley/react-form#14]
Oh hey, me again.

I believe there's an issue with Touched(), the docs say ```When a field is changed via setValue() or setTouched(), its corresponding field location in this object is marked as true.```  However, this isn't always the case, and it's messing with some error handling I'm trying to implement.

It can be shown even on the demo at http://react-form.zabapps.com/

Type "Bob" into the full name field and scroll down to the output of the "Touched" array. Don't click so the box loses focus. Touched is reporting that ```name``` has a touched status of "Bob". Now click somewhere, and the Touched status is now ```true```. 

I'm assuming that's not by design?

**Comment by tannerlinsley** [comment]
You are correct. That's not expected behavior.  This line: https://github.com/tannerlinsley/react-form/blob/master/src/form.js#L69 needs to be changed from:
```javascript
const touched = _.set(state.touched, field, value)
```
to:
```javascript
const touched = _.set(state.touched, field)
```

This way, the value will not be passed, and the `touched` value will default to `true`

If you want to fix it, I'll accept a PR for it :)

**Comment by vladra** [comment]
I have not exactly same issue, but it is somehow related to this line also.

Lets say I have 3 input fields - `user.firstName`, `user.lastName` and `company.name`

After I do this (done on button click) - `setValue('user', {})` - I see strange bug. On the onBlur event (set to `() => setTouched`), input value is set to `true`.

I've found 2 ways of fixing this problem:

1. Doing `setValue('user', {}, true)`
2. Removing `value` from the line in the previous message

I can't understand how it is related though, especially second option. When onBlur event happens, `setValue` is not called at all - but removing `value` from line 69 somehow fixes it.

But removing value from line 69 is not marking field as touched with onChange events.

Also I've tried to replace value with true. It works fine except the case of nested forms. In this case on form reset, I have `{ values: { user: {} }, touched: { user: true } }`

ps I was removing value from "line 69" on slightly different source code, which comes from npm package

any tips are welcome :)
